src/Diagnostics.DataProviders/KustoClient.cs

```diff
@@ -125,10 +127,12 @@ private DataTableResponseObjectCollection ProcessKustoResponse(string responseCo
             return JsonConvert.DeserializeObject<DataTableResponseObjectCollection>(responseContent);
         }
 
-        private void LogKustoQuery(string query, string cluster, string operationName, Stopwatch timeTakenStopWatch, string kustoClientId, Exception kustoApiException, DataTableResponseObjectCollection dataSet)
+        private void LogKustoQuery(string query, string cluster, string operationName, Stopwatch timeTakenStopWatch, string kustoClientId, Exception kustoApiException, DataTableResponseObjectCollection dataSet, string kustoResponse = "")
         {
             var status = kustoApiException == null ? "Success" : "Failed";
 
+            kustoResponse = (kustoResponse != "") ? $" {Environment.NewLine} --> KustoResponseBody : {kustoResponse} " : string.Empty ;
```

Comment:
nitpik: Since you already have this `NewLine` in the beginning maybe you don't need the `-->`. The new line will delimit the exception.ToString() and the response body plus you already prefix kustoResponse with `KustoResponseBody` so we should be able to easily figure out which is the kusto exception